src/slurm.jl

```diff
@@ -34,24 +34,58 @@ function launch(manager::SlurmManager, params::Dict, instances_arr::Array,
                 end
             end
         end
+
+        # rm("job*.out")
         np = manager.np
         jobname = "julia-$(getpid())"
-        srun_cmd = `srun -J $jobname -n $np -D $exehome $(srunargs) $exename $exeflags --worker`
+        srun_cmd = `srun -J $jobname -n $np -o "job%3t.out" -D $exehome $(srunargs) $exename $exeflags --worker`
         out, srun_proc = open(srun_cmd)
-        for i = 1:np
-            w = split(split(readline(out), ":")[2], "#")
-            config = WorkerConfig()
-            config.port = int(w[1])
-            config.host = strip(w[2])
-            # Keep a reference to the proc, so it's properly closed once
-            # the last worker exits.
-            config.userdata = srun_proc
-            push!(instances_arr, config)
-            notify(c)
+        local w
+        for i = 0:np - 1
+            print("connecting to worker $(i + 1) out of $np\r")
+            fn = "$exehome/job$(lpad(i, 3, "0")).out"
+            t0 = time()
+            while true
+                if time() > t0 + 60 + np
+                    warn("dropping worker: file not created in $(60 + np) seconds")
+                    break
+                end
+                sleep(0.001)
+                if isfile(fn)
+                    t0 = time()
+                    while true
+                        if time() > t0 + 60 + np
+                            warn("dropping worker: file created, but not written to in $(60 + np) seconds")
+                            break
+                        end
+                        w = open(fn) do f
+                            rf = readline(f)
+                            if ismatch(r"\n", rf)
+                                return split(split(rf, ":")[2], "#")
+                            end
+                            return ""
+                        end
+                        length(w) > 0 && break
+                        sleep(0.001)
+                    end
+                    break
+                end
+            end
+            isfile(fn) && rm(fn)
+            if length(w) > 0
+                config = WorkerConfig()
+                config.port = parse(Int, w[1])
+                config.host = strip(w[2])
+                # Keep a reference to the proc, so it's properly closed once
+                # the last worker exits.
+                config.userdata = srun_proc
+                push!(instances_arr, config)
+                notify(c)
+            end
```

Comment:
You may get a little more speedup by wrapping individual output file reads in `@async` blocks, within an overall `@sync` block.